### PR TITLE
Adjust the splitChunks

### PR DESCRIPTION
**Pull request by tvkhoa**
https://employmenthero.atlassian.net/browse/KON-1528

The sum of bundle size reduce from **> 16830 KB** to **6173 KB**

The time of build (on local) reduces from  ~ 600s   to  **~300s**

On sandbox: Build step from  ~ 11-12 mins   to  **5-6 mins**

### CODE REVIEW

File: packages/react-scripts/config/webpack.config.js [hunk 1]
```diff
@@ -294,6 +294,23 @@ module.exports = function(webpackEnv) {
       splitChunks: {
         chunks: 'all',
         name: true, // EH Custom
+        minSize: 20000,
+        minChunks: 1,
+        maxAsyncRequests: 30,
+        maxInitialRequests: 30,
+        cacheGroups: {
+          // Turn it on for long term caching
+          // heroDesign: {
```
Comment (tvkhoa) [inline packages/react-scripts/config/webpack.config.js:303]: disable for now due to backward compatible, will enable it if we want the better long term caching

File: packages/react-scripts/config/webpack.config.js [hunk 2]
```diff
@@ -750,14 +767,14 @@ module.exports = function(webpackEnv) {
...
       // EH Custom
-      isEnvProduction && process.env.CHECK_DEADCODE === 'true' && new DeadCodePlugin({
-        patterns: [
-          'src/**/*.(js|jsx|css)',
-        ],
-        exclude: process.env.DEADCODE_EXCLUDE
-          ? process.env.DEADCODE_EXCLUDE.split(',')
-          : ['**/*.(stories|spec).(js|jsx)'],
-      })
+      isEnvProduction &&
```
Comment (tvkhoa) [inline packages/react-scripts/config/webpack.config.js:770]: just new rule of prettier

**Comment by cuongn265** [comment]
Wow, just wonder how some simples LOCs can optimize such a huge performance like that, can you help adding some reference to how it work a?

**Comment by tvkhoa** [comment]
@cuongn265 https://employmenthero.atlassian.net/wiki/spaces/FC/pages/1712881909/Optimize+JS+bundling - you can download the chart before and after to compare. You can see in the before one, the reusable vendors were not optimized (example: velonic is bundled in multiple chunks).
This PR just let the config the webpack to allow it to separate the chunks to the smaller chunks which will let it reuse the vendors more effective.

**Comment by tvkhoa** [comment]
btw, approve this PR pls cc @phthhieu

**Review by phthhieu — APPROVED** [review]

**Pull request merged**